File: BancoPython/BankAccount.py

class BankAccount:
    
    __accountnumber = 0
    __monto = 0.0
    __tipo = ''
    
      

    def __init__(self, account_number, tipo):
        self.account_number = account_number
        self.tipo = tipo
        self.monto = 0.0

    def get_account_number(self):
        return self.account_number

    def set_account_number(self, account_number):
        self.account_number = account_number

    def get_monto(self):
        return self.monto

    def set_monto(self, monto):
        self.monto = monto

    def get_tipo(self):
        return self.tipo

    def set_tipo(self, tipo):
        self.tipo = tipo

    def mostrar_informacion(self):
        print(f"Número de cuenta: {self.account_number}\nSaldo: {self.monto}\nTipo de cuenta: {self.tipo}")

    def depositar(self, cantidad, tipo):
        try:
            if tipo.lower() == 'a':
                self._add_money_a(cantidad)
            elif tipo.lower() == 'b':
                self._add_money_b(cantidad)
            else:
                self.monto += cantidad
                self._deposito_correcto(cantidad)
        except Exception as e:
            print("Introduce una cantidad valida para tu cuenta")

    def _add_money_a(self, cantidad):
        if self.monto + cantidad >= 50000:
            print("No puedes tener más de 50,000 en tipo de cuenta A")
        else:
            self.monto += cantidad
            self._deposito_correcto(cantidad)

    def _add_money_b(self, cantidad):
        if self.monto + cantidad >= 100000:
            print("No puedes tener más de 100,000 en tipo de cuenta B")
        else:
            self.monto += cantidad
            self._deposito_correcto(cantidad)

    def _deposito_correcto(self, cantidad):
        print(f"Se ha depositado correctamente {cantidad} a su cuenta, ahora su saldo es {self.monto}")

    def retirar(self, cantidad, tipo):
        if tipo.lower() == 'a':
            self._retirar_money_a(cantidad)
        elif tipo.lower() == 'b':
            self._retirar_money_b(cantidad)
        elif tipo.lower() == 'c':
            self._retirar_money_c(cantidad)

    def _retirar_money_a(self, cantidad):
        if self.monto - cantidad >= 1000:
            self.monto -= cantidad
            print(f"Se ha retirado correctamente {cantidad} a su cuenta, ahora su saldo es {self.monto}")
        else:
            print("No puedes tener menos de 1000 en tu cuenta tipo A")

    def _retirar_money_b(self, cantidad):
        if self.monto - cantidad >= 5000:
            self.monto -= cantidad
            print(f"Se ha retirado correctamente {cantidad} a su cuenta, ahora su saldo es {self.monto}")
        else:
            print("No puedes tener menos de 5000 en tu cuenta tipo B")

    def _retirar_money_c(self, cantidad):
        if self.monto - cantidad >= 10000:
            self.monto -= cantidad
            print(f"Se ha retirado correctamente {cantidad} a su cuenta, ahora su saldo es {self.monto}")
        else:
            print("No puedes tener menos de 10000 en tu cuenta tipo C")

File: Tarea7_U2/src/tarea7_u2/BankAccount.java

package tarea7_u2;

public class BankAccount {
    Employee empleado;
    private long accountNumber;
    private double monto;
    public char tipo;

    public BankAccount(long accountNumber, char tipo) {
        this.accountNumber = accountNumber;
        this.tipo = tipo;
    }

    public long getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(long accountNumber) {
        this.accountNumber = accountNumber;
    }

    public double getMonto() {
        return monto;
    }

    public void setMonto(double monto) {
        this.monto = monto;
    }

    public char getTipo() {
        return tipo;
    }

    public void setTipo(char tipo) {
        this.tipo = tipo;
    }
    
    public void mostrarInformacion() {
        System.out.println("Número de cuenta: " + accountNumber + "\nSaldo: " + monto + "\nTipo de cuenta: " + tipo);
    }

    
    public void depositar (double cantidad, char tipo){
        try{
        if( tipo == 'A' || tipo == 'a') addMoneyA(cantidad);
        else if( tipo == 'B' || tipo == 'b') addMoneyB(cantidad);
        else this.monto += cantidad; 
        }catch(Exception e){
            System.out.println("Introduce una cantidad valida para tu cuenta");
        }
    }
    private void addMoneyA(double cantidad){
        if(this.monto + cantidad >= 50000){
            System.out.println("No puedes tener mas de 50 000 en tipo de cuenta A");
            
        }
        else{
            monto = monto + cantidad;
            System.out.println("Se ha depositado correctamente "+cantidad+ " a su cuenta, ahora su saldo es "+this.monto);
            
        }
    }
    private void addMoneyB(double cantidad){
        if(this.monto + cantidad >= 100000){
            System.out.println("No puedes tener mas de 100 000 en tipo de cuenta B");
            
        }
        else{
           
            monto = monto + cantidad;
            System.out.println("Se ha depositado correctamente "+cantidad+ " a su cuenta, ahora su saldo es "+this.monto);
        }
    }
    public void retirar (double cantidad, char tipo){
        if( tipo == 'A'|| tipo == 'a')retirarMoneyA (cantidad);
        else if( tipo == 'B'|| tipo == 'b')retirarMoneyB (cantidad);
        else if( tipo == 'C'|| tipo == 'c')retirarMoneyC (cantidad);
           
    }
    private void retirarMoneyA (double cantidad){
        if( this.monto - cantidad >= 1000 ){
           monto = monto-cantidad;
           System.out.println("Se ha retirado correctamente "+cantidad+ " a su cuenta, ahora su saldo es "+this.monto);
        }
        else 
            System.out.println("No puedes tener menos de 1000 en tu cuenta tipo A");
        }
    private void retirarMoneyB (double cantidad){
        if( this.monto - cantidad >= 5000 ){
           monto = monto-cantidad;
           System.out.println("Se ha retirado correctamente "+cantidad+ " a su cuenta, ahora su saldo es "+this.monto);
        }
        else 
            System.out.println("No puedes tener menos de 5000 en tu cuenta tipo B");
        }
    private void retirarMoneyC (double cantidad){
        if(this.monto - cantidad >= 10000 ){
           monto = monto-cantidad;
           System.out.println("Se ha retirado correctamente "+cantidad+ " a su cuenta, ahora su saldo es "+this.monto);
        }
        else 
            System.out.println("No puedes tener menos de 10000 en tu cuenta tipo C");
        }
    
    private void depositoCorrecto (double cantidad){
        System.out.println("Se ha depositado correctamente" +cantidad+ " a su cuenta, ahora su saldo es"+this.monto);
    }

}
